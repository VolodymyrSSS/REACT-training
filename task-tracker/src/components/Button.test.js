import React from 'react';
import renderer from 'react-test-renderer'; // ES6
import Button from './Button';

describe('<Button />', () => {
  it('should render a button with a className', () => {
    const tree = renderer.create(
      <Button></Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should attached an onclick handler passed in', () => {
    const tree = renderer.create(
      <Button onClick={() => {}}></Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render its children', () => {
    const tree = renderer.create(
      <Button>Some text</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})