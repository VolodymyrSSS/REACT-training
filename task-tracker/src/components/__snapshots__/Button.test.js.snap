// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should attached an onclick handler passed in 1`] = `
<button
  className="btn"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "steelBlue",
    }
  }
/>
`;

exports[`<Button /> should render a button with a className 1`] = `
<button
  className="btn"
  style={
    Object {
      "backgroundColor": "steelBlue",
    }
  }
/>
`;

exports[`<Button /> should render its children 1`] = `
<button
  className="btn"
  style={
    Object {
      "backgroundColor": "steelBlue",
    }
  }
/>
`;
